src/main/java/nextstep/subway/entity/Sections.java

```diff
@@ -0,0 +1,177 @@
+        validateStationDuplication(section.getUpStation());
+        List<Section> sections = new ArrayList<>(sectionList);
+        sectionList.clear();
+        sectionList.add(section);
+        for (Section s : sections) {
+            sectionList.add(new Section(s.getLine(), s.getUpStation(), s.getDownStation(), s.getDistance()));
+        }
+    }
+
+    private void addSectionToEnd(Section section) {
+        validateStationDuplication(section.getDownStation());
+        sectionList.add(section);
+    }
+
+    private void addSectionToMiddle(Section section) {
+        Station newUpStation = section.getUpStation();
+        Station newDownStation = section.getDownStation();
+        Long newDistance = section.getDistance();
+
+        int oldIndex = 0;
+        Section oldSection = new Section();
+        for (int index = 0; index < sectionList.size(); index++) {
+            Section currentSection = sectionList.get(index);
+            if (currentSection.getUpStation().getId() == newUpStation.getId()) {
+                validateStationDuplication(newDownStation);
+                validateDistance(currentSection, newDistance);
+                oldIndex = index;
+                oldSection = currentSection;
+                break;
+            }
+        }
+        Section rightSection = new Section(section.getLine(), newDownStation, oldSection.getDownStation(),
+                oldSection.getDistance() - newDistance);
+        sectionList.set(oldIndex, rightSection);
+
+        Section leftSection = new Section(section.getLine(), newUpStation, newDownStation, newDistance);
+        sectionList.add(oldIndex, leftSection);
+    }
+
+    private void validateDistance(Section currentSection, Long newDistance) {
+        if (currentSection.getDistance() <= newDistance || newDistance <= 0) {
+            throw new IllegalSectionException("신규 구간 거리가 올바르지 않습니다.");
+        }
+    }
+
+    private boolean isMiddleSection(Station newSectionUpStation) {
+        return sectionList.stream()
+                .anyMatch(sec -> sec.getUpStation().getId() == newSectionUpStation.getId());
+    }
+
+    public void removeSection(Station downStation) {
+
+        validateDeleteEmpty();
+
+        validateDeleteOnlyOne();
+
+        validateDeletableSection(downStation);
+
+        sectionList.remove(sectionList.size() - 1);
+    }
+
+    private Optional<Section> findNextSection(Section tempSection) {
+        return sectionList.stream()
+                .filter(section -> section.getUpStation().equals(tempSection.getDownStation()))
+                .findFirst();
+    }
+
+    private void validateStationDuplication(Station station) {
+        for (Section s : sectionList) {
+            if (s.getUpStation().getId() == station.getId()
+                    || s.getDownStation().getId() == station.getId()) {
+                throw new IllegalSectionException("이미 노선에 등록되어있는 역입니다.");
+            }
+        }
```

Comment:
여기도 indent가 깊어보이네요! 
그리고 구간정보 엔티티의 내부 값을 꺼내어 직접 연산을 하는데, 객체지향 패러다임상 객체간 요청과 응답에는 맞지 않는 것 같아요. 
다음과 같이 리팩터링 해볼 수 있지 않을까요?
```suggestion
        boolean isContain = sectionList.stream().anyMatch(section -> section.containsStation(station));
        if (isContain) {
            throw 예외
        }
```